python/lsst/pipe/tasks/healSparseMappingProperties.py

```diff
@@ -0,0 +1,490 @@
+        Number of sampling points in the y direction.
+    orderx : `int`, optional
+        Chebyshev polynomial order for fit in x direction.
+    ordery : `int`, optional
+        Chebyshev polynomial order for fit in y direction.
+
+    Returns
+    -------
+    psf_array : `np.ndarray`
+        Record array with "psf_size", "psf_e1", "psf_e2".
+    """
+    pts = [lsst.geom.SpherePoint(r*lsst.geom.degrees, d*lsst.geom.degrees) for
+           r, d in zip(ra, dec)]
+    pixels = ccd_row.getWcs().skyToPixel(pts)
+
+    ctrl = ChebyshevBoundedFieldControl()
+    ctrl.orderX = orderx
+    ctrl.orderY = ordery
+    ctrl.triangular = False
+
+    bbox = ccd_row.getBBox()
+    xSteps = np.linspace(bbox.getMinX(), bbox.getMaxX(), nx)
+    ySteps = np.linspace(bbox.getMinY(), bbox.getMaxY(), ny)
+    x = np.tile(xSteps, nx)
+    y = np.repeat(ySteps, ny)
+
+    psf_size = np.zeros(x.size)
+    psf_e1 = np.zeros(x.size)
+    psf_e2 = np.zeros(x.size)
+    psf_area = np.zeros(x.size)
+
+    psf = ccd_row.getPsf()
+    for i in range(x.size):
+        shape = psf.computeShape(lsst.geom.Point2D(x[i], y[i]))
+        psf_size[i] = shape.getDeterminantRadius()
+        ixx = shape.getIxx()
+        iyy = shape.getIyy()
+        ixy = shape.getIxy()
+
+        psf_e1[i] = (ixx - iyy)/(ixx + iyy + 2.*psf_size[i]**2.)
+        psf_e2[i] = (2.*ixy)/(ixx + iyy + 2.*psf_size[i]**2.)
+
+        im = psf.computeKernelImage(lsst.geom.Point2D(x[i], y[i]))
+        psf_area[i] = np.sum(im.array)/np.sum(im.array**2.)
+
+    pixel_x = np.array([pix.getX() for pix in pixels])
+    pixel_y = np.array([pix.getY() for pix in pixels])
+
+    psf_array = np.zeros(pixel_x.size, dtype=[("psf_size", "f8"),
+                                              ("psf_e1", "f8"),
+                                              ("psf_e2", "f8"),
+                                              ("psf_area", "f8")])
+
+    cheb_size = ChebyshevBoundedField.fit(lsst.geom.Box2I(bbox), x, y, psf_size, ctrl)
+    psf_array["psf_size"] = cheb_size.evaluate(pixel_x, pixel_y)
+    cheb_e1 = ChebyshevBoundedField.fit(lsst.geom.Box2I(bbox), x, y, psf_e1, ctrl)
+    psf_array["psf_e1"] = cheb_e1.evaluate(pixel_x, pixel_y)
+    cheb_e2 = ChebyshevBoundedField.fit(lsst.geom.Box2I(bbox), x, y, psf_e2, ctrl)
+    psf_array["psf_e2"] = cheb_e2.evaluate(pixel_x, pixel_y)
+    cheb_area = ChebyshevBoundedField.fit(lsst.geom.Box2I(bbox), x, y, psf_area, ctrl)
+    psf_array["psf_area"] = cheb_area.evaluate(pixel_x, pixel_y)
+
+    return psf_array
+
+
+class PropertyMapMap(dict):
+    """Map of property maps to be run for a given task.
+
+    Notes
+    -----
+    Property maps are classes derived from `BasePropertyMap`
+    """
+    def __iter__(self):
+        for property_map in self.values():
+            if property_map.config.do_min or property_map.config.do_max \
+               or property_map.config.do_mean or property_map.config.do_weighted_mean \
+               or property_map.config.do_sum:
```

Comment:
If you do parentheses then the indentation is too large and that's a flake8 violation.  If you know how to get parentheses to work here then I'm all ears.  (Rather than putting double parentheses but I don't know if that's an improvement over the back-slashes.)

Reply:
`            if (property_map.config.do_min or property_map.config.do_max or property_map.config.do_mean
                    or property_map.config.do_weighted_mean or property_map.config.do_sum):`